legate/driver/launcher.py

```diff
@@ -186,16 +187,18 @@ def _compute_env(self) -> tuple[EnvDict, set[str]]:
         # threading support
         env["GASNET_MPI_THREAD"] = "MPI_THREAD_MULTIPLE"
 
-        # UCX-related environment variables
-        env["UCX_CUDA_COPY_MAX_REG_RATIO"] = "1.0"
-        env["UCX_MULTI_LANE_MAX_RATIO"] = "1.0"
-        env["UCX_IB_RCACHE_PURGE_ON_FORK"] = "n"
-        env["UCX_RC_TX_POLL_ALWAYS"] = "y"
-
-        # Link to the UCX bootstrap plugin, in case Realm is using UCX
-        env["REALM_UCP_BOOTSTRAP_PLUGIN"] = str(
-            system.legion_paths.legion_lib_path / "realm_ucp_bootstrap_mpi.so"
-        )
+        if config.multi_node.ranks > 1 and "ucx" in install_info.networks:
+            # UCX-related environment variables
+            env["UCX_CUDA_COPY_MAX_REG_RATIO"] = "1.0"
+            env["UCX_MULTI_LANE_MAX_RATIO"] = "1.0"
+            env["UCX_IB_RCACHE_PURGE_ON_FORK"] = "n"
+            env["UCX_RC_TX_POLL_ALWAYS"] = "y"
+
+            # Link to the UCX bootstrap plugin
+            env["REALM_UCP_BOOTSTRAP_PLUGIN"] = str(
+                system.legion_paths.legion_lib_path
+                / "realm_ucp_bootstrap_mpi.so"
+            )
```

Comment:
Maybe time to think about splitting `_compute_env` up into `env_nccl`, `env_ucc`, etc but no need to do that in this PR.